lib/Bindings/Python/circt/dialects/_comb_ops_ext.py

```diff
@@ -0,0 +1,110 @@
+from circt.dialects import comb
+from circt.support import NamedValueOpView
+
+from mlir.ir import IntegerAttr, IntegerType
+
+
+# Builder base classes for non-variadic unary and binary ops.
+class UnaryOpBuilder(NamedValueOpView):
+
+  def operand_names(self):
+    return ["input"]
+
+  def result_names(self):
+    return ["result"]
+
+
+class ExtractOpBuilder(UnaryOpBuilder):
+
+  def __init__(self, low_bit, data_type, input_port_mapping={}, **kwargs):
+    low_bit = IntegerAttr.get(IntegerType.get_signless(32), low_bit)
+    super().__init__(comb.ExtractOp, data_type, input_port_mapping, [],
+                     [low_bit], **kwargs)
+
+
+class BinaryOpBuilder(NamedValueOpView):
+
+  def operand_names(self):
+    return ["lhs", "rhs"]
+
+  def result_names(self):
+    return ["result"]
+
+
+# Sugar classes for the various non-variadic unary ops.
+class ExtractOp:
+
+  @staticmethod
+  def create(low_bit, *args, **kwargs):
+    return ExtractOpBuilder(low_bit, *args, **kwargs)
+
+
+class ParityOp:
```

Comment:
A lot of these are quite similar... Is there a reason why these don't just subclass something?

Reply:
Or have a class annotation?

Reply:
It should be safe to just subclass something here.

Reply:
Ended up going with annotations, since that is a nice way to parameterize the comparator ops.